### PR TITLE
NDRS-524: convert tests to use reactor macro

### PR DESCRIPTION
**Pull request by marc-casperlabs**
Conversion of fetcher tests using the new reactor model.

Currently, this has two flaws:

1. It is somewhat complicated to inject a `Chainspec` as it was done in the old implementation of the tests. This is not an issue of the macro itself, but rather something it exposes. It could be worked around using delegated dispatch (see below), but overall there must be a cleaner strategy here.

2. This gives devs enough rope to hang themselves again with the `fn my_dispatch` escape hatch. At the moment I cannot think of a better alternative yet that does not require refactoring.

### CODE REVIEW

File: node/src/components/deploy_acceptor/event.rs [hunk 1]
```diff
@@ -28,6 +28,17 @@ pub enum Event {
     },
 }
 
+impl From<ApiServerAnnouncement> for Event {
+    fn from(announcement: ApiServerAnnouncement) -> Self {
+        match announcement {
+            ApiServerAnnouncement::DeployReceived { deploy } => Event::Accept {
+                deploy: deploy,
```
Comment (EdHastingsCasperLabs) [inline node/src/components/deploy_acceptor/event.rs:35]:
```suggestion
                deploy,
```

File: node/src/components/in_memory_network.rs [hunk 2]
```diff
@@ -303,12 +303,32 @@ use crate::{
...
 };
+use fmt::Formatter;
 
 /// The node ID type used by the in-memory network.
 pub type NodeId = KeyFingerprint;
 
+/// A network.
 type Network<P> = Arc<RwLock<HashMap<NodeId, mpsc::UnboundedSender<(NodeId, P)>>>>;
 
+/// Public type alias for convention; the in memory network needs to internal events and just
```
[the remark this answers is not in the document]
Comment (marc-casperlabs) [reply node/src/components/in_memory_network.rs:312]: The comment is stale back when this was an alias. It is now a proper newtype.

File: node/src/effect/announcements.rs [hunk 3]
```diff
@@ -57,6 +57,10 @@ where
 }
 
 /// An HTTP API server announcement.
+// TODO: This needs to renamed and factored out -- new incoming deploys should be their own
```
[the remark this answers is not in the document]
Comment (marc-casperlabs) [reply node/src/effect/announcements.rs:61]: The source component of a new deploy (or anything, really) must be irrelevant. Announcements are supposed to decouple components.

Unfortunately we have quite a few `XComponentAnnouncements`, but these should be more accurately named after what they are. It is helpful to look at announcements in a "bottom-up" way: Every `enum`-Variant of announcements initially should be its own type and not related to the component that emits it (after all, others could emit it, too).

The fact that sometimes these announcements are bundled up in an `enum` is for mere convenience, as in some cases, there are almost no cases in which one would not handle a complete set of announcements (or requests).

Ultimately the fact that multiple of these stem from a single component in practice should not cloud this view though.

**Comment by marc-casperlabs** [comment]
It's finally green :sweat_smile: (at least locally)

I did a tiny refactoring, removing errors from the chainspec constructor (there aren't any anymore). The chainspec loader is now used in the tests, instead of a shim, to commit it to storage. Genesis commitment is ignored.

**Review by Fraser999 — APPROVED** [review]

File: node/src/components/fetcher/tests.rs [hunk 4]
```diff
@@ -1,276 +1,158 @@
...
+        FetcherRequest<NodeId, Deploy> -> deploy_fetcher;
+    }
 
-        Ok((reactor, effects))
+    announcements: {
+        // The deploy fetcher needs to be notified about new deploys.
+        DeployAcceptorAnnouncement<NodeId> -> [deploy_fetcher];
+        NetworkAnnouncement<NodeId, Message> -> [fn handle_message];
+        // Currently the RpcServerAnnouncement is misnamed - it solely tells of new deploys arriving
+        // from a client.
```
[the remark this answers is not in the document]
Comment (marc-casperlabs) [reply node/src/components/fetcher/tests.rs:94]: Alright, for now :)

**Review by EdHastingsCasperLabs — APPROVED** [review]
Commitapalooza!

**Pull request merged**